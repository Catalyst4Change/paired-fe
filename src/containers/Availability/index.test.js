import React from 'react';
import { shallow } from 'enzyme';
import { Availability } from './index';

describe('Availability', () => {
  let wrapper;
  let mockHandleClick = jest.fn();
  let mockAvailabilities = [true, false, true, false, true, false];

  beforeEach(() => {
    wrapper = shallow(
      <Availability
        handleClick={mockHandleClick}
        availabilities={mockAvailabilities}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
