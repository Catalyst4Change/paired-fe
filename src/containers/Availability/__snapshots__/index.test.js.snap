// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Availability Availability component should match the snapshot when the path is /edit-availability 1`] = `
<section
  className="Availability"
>
  <h2
    className="Availability--h2"
  >
    When are you available to help to others?
  </h2>
  <table
    className="Availability--table"
  >
    <thead>
      <tr>
        <th />
        <th>
          8:00 - 8:30 a.m.
        </th>
        <th>
          12:00 - 12:30 p.m.
        </th>
        <th>
          4:10 - 4:40 p.m.
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="row-name"
        >
          Mon
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="0"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="1"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="2"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Tue
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="3"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="4"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="5"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Wed
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="6"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="7"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="8"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Thu
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="9"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="10"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="11"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Fri
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="12"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="13"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="14"
            onClick={[Function]}
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="Availability--div-buttons"
  >
    <button
      id="cancel--button"
      onClick={[Function]}
    >
      Cancel
    </button>
    <button
      id="edit--button"
      onClick={[Function]}
    >
      Edit Availability
    </button>
  </div>
</section>
`;

exports[`Availability Availability component should match the snapshot when the path is /set-availability 1`] = `
<section
  className="Availability"
>
  <h2
    className="Availability--h2"
  >
    When are you available to help to others?
  </h2>
  <table
    className="Availability--table"
  >
    <thead>
      <tr>
        <th />
        <th>
          8:00 - 8:30 a.m.
        </th>
        <th>
          12:00 - 12:30 p.m.
        </th>
        <th>
          4:10 - 4:40 p.m.
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="row-name"
        >
          Mon
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="0"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="1"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="2"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Tue
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="3"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="4"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="5"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Wed
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="6"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="7"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="8"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Thu
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="9"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="10"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="11"
            onClick={[Function]}
          />
        </td>
      </tr>
      <tr>
        <td
          className="row-name"
        >
          Fri
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="12"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="13"
            onClick={[Function]}
          />
        </td>
        <td>
          <button
            className="button--availability button--false"
            key="14"
            onClick={[Function]}
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="Availability--div-buttons"
  >
    <button
      id="skip--button"
      onClick={[Function]}
    >
      Skip
    </button>
    <button
      id="set--button"
      onClick={[Function]}
    >
      Set Availability
    </button>
  </div>
</section>
`;
